sdcm/utils/file.py

```diff
@@ -91,18 +103,20 @@ def readline(self, limit: int = -1) -> AnyStr:
     def readlines(self, hint: int = -1) -> List[AnyStr]:
         return self._io.readlines(hint)
 
-    def read_lines_filtered(self, *patterns: Union[Pattern]) -> List[str]:
+    def read_lines_filtered(self, *patterns: Union[Pattern]) -> Iterable[str]:
         """
         Read lines from the file, filter them and yield
         :param patterns: List of patterns
         :param return_pattern: If True, it will return tuple of string and pattern
         :return:
         """
-        for line in self._io:
-            for pattern in patterns:
-                if pattern.search(line):
-                    yield line
-                    break
+        def generator():
+            for line in self._io:
+                for pattern in patterns:
+                    if pattern.search(line):
+                        yield line
+                        break
+        return ReiterableGenerator(generator=generator)
```

Comment:
why do we need to reiterate?

Reply:
To be able to re iterate result of read_lines_filtered again. For example when you read logs in cycle.

Reply:
please check `itertools.cycle`